Gems/ROS2/Code/Source/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp

```diff
@@ -61,4 +75,92 @@ namespace ROS2
+        auto visualNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), true, false);
+        auto meshNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), true, true);
+        AZStd::shared_ptr<Utils::UrdfAssetMap> urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>();
+        if (importAssetWithUrdf)
+        {
+            urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>(
+                Utils::CopyAssetForURDFAndCreateAssetMap(meshNames, filePath, collidersNames, visualNames));
+        }
+        bool allAssetProcessed = false;
+        do
+        {
+            allAssetProcessed = true;
+            for (const auto& [name, asset] : *urdfAssetsMapping)
+            {
+                auto sourceAssetFullPath = asset.m_availableAssetInfo.m_sourceAssetGlobalPath;
+                if (sourceAssetFullPath.empty())
+                {
+                    AZ_Warning("ROS2EditorSystemComponent", false, "asset %s is has no sourceAssetFullPath", name.c_str());
+                    continue;
+                }
+                using namespace AzToolsFramework;
+                using namespace AzToolsFramework::AssetSystem;
+                AZ::Outcome<AssetSystem::JobInfoContainer> result = AZ::Failure();
+                AssetSystemJobRequestBus::BroadcastResult(
+                    result, &AssetSystemJobRequestBus::Events::GetAssetJobsInfo, sourceAssetFullPath, true);
+                JobInfoContainer& allJobs = result.GetValue();
+                for (const JobInfo& job : allJobs)
+                {
+                    if (job.m_status == JobStatus::Queued || job.m_status == JobStatus::InProgress)
+                    {
+                        AZ_Printf("ROS2EditorSystemComponent", "asset %s is being processed", sourceAssetFullPath.c_str());
+                        allAssetProcessed = false;
+                    }
+                    else
+                    {
+                        AZ_Printf("ROS2EditorSystemComponent", "asset %s is done", sourceAssetFullPath.c_str());
+                    }
+                }
+            }
+
+            if (allAssetProcessed)
+            {
+                AZ_Printf("ROS2EditorSystemComponent", "All assets processed");
+            }
+        } while (!allAssetProcessed);
+
+        AZStd::string prefabName = AZStd::string(parsedUrdf->getName().c_str(), parsedUrdf->getName().size()) + ".prefab";
+
+        const AZ::IO::Path prefabPathRelative(AZ::IO::Path("Assets") / "Importer" / prefabName);
+        const AZ::IO::Path prefabPath(AZ::IO::Path(AZ::Utils::GetProjectPath()) / prefabPathRelative);
+        AZStd::unique_ptr<URDFPrefabMaker> prefabMaker;
+        prefabMaker = AZStd::make_unique<URDFPrefabMaker>(filePath, parsedUrdf, prefabPath.String(), urdfAssetsMapping, useArticulation);
+
+        auto prefabOutcome = prefabMaker->CreatePrefabFromURDF();
+
+        if (!prefabOutcome.IsSuccess())
+        {
+            return false;
```

Comment:
Maybe add an `AZ_Error` statement here?
```suggestion
            AZ_Error("ROS2EditorSystemComponent", false, "Unable to create Prefab from URDF file %s", filePath.data());
            return false;
```